tools/toolchains/arm.py

```diff
@@ -80,6 +80,13 @@ def __init__(self, target, notify=None, macros=None,
                 self.flags['ld'].append("--library_type=microlib")
             if "--library_type=microlib" not in self.flags['common']:
                 self.flags['common'].append("--library_type=microlib")
+        
+        stdio_lib = "std"
+        if hasattr(target, "stdio_lib"):            
+            stdio_lib = target.stdio_lib
+
+        if stdio_lib == "small":
```

Comment:
```suggestion
        if hasattr(target, "stdio_lib") and target.stdio_lib == "small":
            self.flags["common"].append("-DMBED_MINIMAL_PRINTF")
```

`stdio_lib` and `std` are not needed. The special case is if `target.stdio_lib` is `small`.